controllers/datadogagent/override/global.go

```diff
@@ -6,34 +6,220 @@
-func ApplyGlobalSettings(manager feature.PodTemplateManagers, config *v2alpha1.GlobalConfig) *corev1.PodTemplateSpec {
-	// TODO(operator-ga): implement ApplyGlobalSettings
-	if config != nil && config.ClusterName != nil {
+func ApplyGlobalSettings(logger logr.Logger, manager feature.PodTemplateManagers, dda *v2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, componentName v2alpha1.ComponentName) *corev1.PodTemplateSpec {
+	var err error
+	config := dda.Spec.Global
+
+	// ClusterName sets a unique cluster name for the deployment to easily scope monitoring data in the Datadog app.
+	if config.ClusterName != nil {
 		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
 			Name:  apicommon.DDClusterName,
 			Value: *config.ClusterName,
 		})
 	}
 
-	if config != nil && config.Kubelet != nil && config.Kubelet.TLSVerify != nil {
-		kubeletEnvVar := &corev1.EnvVar{
-			Name:  apicommon.DDKubeletTLSVerify,
-			Value: apiutils.BoolToString(config.Kubelet.TLSVerify),
+	// Site is the Datadog intake site Agent data are sent to.
+	manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+		Name:  apicommon.DDSite,
+		Value: *config.Site,
+	})
+
+	// Endpoint is the Datadog intake URL the Agent data are sent to.
+	if config.Endpoint != nil && config.Endpoint.URL != nil {
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  apicommon.DDddURL,
+			Value: *config.Endpoint.URL,
+		})
+	}
+
+	// Registry is the image registry to use for all Agent images.
+	for _, c := range manager.PodTemplateSpec().Spec.InitContainers {
+		c.Image = *config.Registry
+	}
+	for _, c := range manager.PodTemplateSpec().Spec.Containers {
+		c.Image = *config.Registry
+	}
+
+	// LogLevel sets logging verbosity. This can be overridden by container.
+	manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+		Name:  apicommon.DDLogLevel,
+		Value: *config.LogLevel,
+	})
+
+	// NetworkPolicy contains the network configuration.
+	if config.NetworkPolicy != nil {
+		if apiutils.BoolValue(config.NetworkPolicy.Create) {
+			switch config.NetworkPolicy.Flavor {
+			case v2alpha1.NetworkPolicyFlavorKubernetes:
+				err = resourcesManager.NetworkPolicyManager().AddKubernetesNetworkPolicy(component.BuildKubernetesNetworkPolicy(dda, componentName))
+			case v2alpha1.NetworkPolicyFlavorCilium:
+				// TODO
+			}
+			if err != nil {
+				logger.Info("Error adding Network Policy to the store", "error", err)
+			}
 		}
-		manager.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, kubeletEnvVar)
-		manager.EnvVar().AddEnvVarToContainer(apicommonv1.ProcessAgentContainerName, kubeletEnvVar)
 	}
 
-	// set image registry
+	if componentName == v2alpha1.NodeAgentComponentName {
+		// Tags contains a list of tags to attach to every metric, event and service check collected.
+		if config.Tags != nil {
+			tags, marshalErr := json.Marshal(config.Tags)
+			if marshalErr != nil {
+				logger.Info("Failed to unmarshal json input", "error", err)
```

Comment:
This one should be `marshalErr` instead of `err`. Same for the other 2 `json.Marshal` calls below.